contracts/Subscription.sol

```diff
@@ -1,205 +1,356 @@
 
-    uint64 private _minimumRenewalDuration;
-    uint64 private _maximumRenewalDuration;
-
-    // crea8ors contract
-    IERC721 public erc721;
+    /*//////////////////////////////////////////////////////////////
+                             PRIVATE STORAGE
+    //////////////////////////////////////////////////////////////*/
 
     bool private _renewable;
 
-    constructor(IERC721 erc721_) {
-        erc721 = erc721_;
-    }
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
 
-    fallback(bytes calldata input) external payable returns (bytes memory) {
-        // console2.log("msg.sender: ", msg.sender);
-        // console2.logBytes(input);
+    /// @dev Initializes the Subscription contract with the address of the NFT and the minter contract addresses.
+    /// @param _cre8orsNFT The address of the NFT contract. (cre8ors)
+    /// @param _minter The address of the minter. (FriendsAndFamilyMinter)
+    constructor(address _cre8orsNFT, address _minter) ERC5643(_cre8orsNFT, _minter) { }
 
-        // send / transfer (forwards 2300 gas to this fallback function)
-        // call (forwards all of the gas)
+    /*//////////////////////////////////////////////////////////////
+                     USER-FACING CONSTANT FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
 
-        // check that caller is erc721 otherwise revert
-        if (msg.sender != address(erc721)) revert CallerNotERC721();
+    /// @inheritdoc ISubscription
+    function isSubscriptionValid(uint256 tokenId) public view override returns (bool) {
+        return expiresAt(tokenId) > block.timestamp;
+    }
 
-        // check data length
-        if (input.length == 0) revert CalledWithOutData();
+    /// @inheritdoc ISubscription
+    function validateSubscription(uint256 tokenId) public view override returns (bool) {
+        bool isValid = isSubscriptionValid(tokenId);
 
-        // console2.log("here");
+        if (!isValid) {
+            revert InvalidSubscription();
+        }
 
-        // Decode the msg.data/input and extract the parameter
-        uint256 tokenId = abi.decode(input, (uint256));
-        // console2.log("tokenId: ", tokenId);
+        return isValid;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                   USER-FACING NON-CONSTANT FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc ISubscription
+    function renewSubscriptionWithERC20Payment(
+        uint256 tokenId,
+        uint64 duration,
+        address erc20
+    )
+        external
+        payable
+        override
+    {
+        _validateCallerAsOwnerOrApproved({ caller: msg.sender, tokenId: tokenId });
+
+        _handleERC20PaymentSubscriptionUpdate(tokenId, duration, erc20);
+    }
```

Comment:
I think we should remove ERC20 methods for simpler contracts & less expensive deployment costs.

what do you think?